add-permissions.js

```diff
@@ -39,8 +39,15 @@ module.exports = class AwsAddLambdaAccountPermissions {
       const functionLogicalId = this.provider.naming.getLambdaLogicalId(functionName);
 
       functionAllowAccess.reduce((previousResourceName, principal) => {
-        principal = principal.toString();
-        const principalName = principal.replace(/\b\w/g, l => l.toUpperCase()).replace(/[_\W]+/g, "");
+        let principalString;
+        if (principal instanceof Object && principal['Fn::ImportValue']) {
```

Comment:
Technically, it can be any CF function, not only `ImportValue`. Ideas on making this generic?